fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java

```diff
@@ -128,7 +130,21 @@ private static void write(final RdfStream rdfStream,
             final StreamRDF stream = getWriterStream(output, format);
             stream.start();
             nsPrefixes.forEach(stream::prefix);
-            rdfStream.forEach(stream::triple);
+
+            // ensure that writes to the stream object are serialized
+            // in case the rdfStream object is running in parallel mode.
+            final ReentrantLock lock = new ReentrantLock();
+            rdfStream.forEach(x -> {
+                try {
+                    lock.tryLock(30, TimeUnit.SECONDS);
```

Comment:
If you're just trying to keep a single thread at a time running through the `StreamRDF`, why not just subtype it with `synchronize void triple(Triple t)`? No need to explicitly manage a lock.